File: AutoTrade/AutoTrade.Services/Database/MoodTracker.cs


using AutoTrade.Services.Database;

namespace Database
{
    public class MoodTracker
    {

        public int Id { get; set; }

        public int UserId { get; set; }
        public User User { get; set; }
        public string Mood { get; set; }
        public string Description { get; set; }
        public DateTime MoodDate { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: AutoTrade/Ui/mobile/vroom_app/lib/models/moodTracker.dart
import 'package:vroom_app/models/user.dart';

class MoodTracker {
  final int id;
  final int? userId;
  final User? user;
  final String mood;
  final String description;
  final DateTime moodDate;
  final DateTime? createdAt;

  MoodTracker(
      {required this.id,
      this.userId,
      this.user,
      required this.mood,
      required this.description,
      required this.moodDate,
      this.createdAt});

  factory MoodTracker.fromJson(Map<String, dynamic> json) {
    return MoodTracker(
        id: json['id'] ?? 0,
        userId: json['userId'] ?? 0,
        user: json['user'] != null ? User.fromJson(json['user']) : null,
        mood: json['mood'],
        description: json['description'],
        moodDate: DateTime.parse(json['moodDate']),
        createdAt: DateTime.parse(json['createdAt']));
  }
}
